devices/temp_nix.go

```diff
@@ -28,6 +32,45 @@ func getTemps(temps map[string]int) map[string]error {
 			temps[label] = int(sensor.Temperature)
 		}
 	}
+
+	block, err := ghw.Block()
+	if err != nil {
+		log.Print("error getting block device info")
+		return nil
+	}
+
+	for _, disk := range block.Disks {
+		dev, err := smart.Open("/dev/" + disk.Name)
```

Comment:
btw there is an opportunity for optimization here. You can open the devices only once, at the application start and keep it as an array of smart.dev objects. When you need the temperature you just iterate over the list of devices and get the required temperature.

Reply:
@anatol something like this? [e96355d](https://github.com/xxxserxxx/gotop/pull/221/commits/e96355d0b82f4075abe059d94c99ce7eaa6a7946)